app/bundles/LeadBundle/Segment/Decorator/CompanyDecorator.php

```diff
@@ -15,22 +15,26 @@
 use Mautic\LeadBundle\Segment\Query\Filter\ComplexRelationValueFilterQueryBuilder;
 
 /**
- * Class CustomMappedDecorator.
+ * Class CompanyDecorator.
  */
 class CompanyDecorator extends BaseDecorator
 {
     /**
+     * @param ContactSegmentFilterCrate $contactSegmentFilterCrate
+     *
      * @return string
      */
-    public function getRelationJoinTable()
+    public function getRelationJoinTable(ContactSegmentFilterCrate $contactSegmentFilterCrate)
```

Comment:
Adding a required method param is a BC break. Isn't there another way to do it? @Maxell92 can you give us a hand as the segment master?

Reply:
Yes, this is true. There is a possibility to create a new decorator for this case, but I am not sure it will be enough.